apps/shared/avifpng.c

```diff
@@ -196,6 +198,65 @@ static avifBool avifExtractExifAndXMP(png_structp png, png_infop info, avifBool
     return AVIF_TRUE;
 }
 
+static float avifToLinear(float v, float gamma)
+{
+    return powf(v, gamma);
+}
+
+#define SRGB_GAMMA_SLOPE 12.92f
+static float avifToSrgbGamma(float linear)
+{
+    static const float kThreshold = 0.04045f / SRGB_GAMMA_SLOPE;
+    if (linear <= kThreshold) {
+        return linear * SRGB_GAMMA_SLOPE;
+    } else {
+        return powf(linear, 1.f / 2.4f) * 1.055f - 0.055f;
+    }
+}
+
+static int avifCorrectGamma(uint16_t v, float gamma, int rgbMaxChannel)
+{
+    const float normalized = v / (float)rgbMaxChannel;
+    const float corrected = avifToSrgbGamma(avifToLinear(normalized, gamma));
+    const int scaledUp = (int)floorf(corrected * rgbMaxChannel + 0.5f);
+    return scaledUp < 0 ? 0 : scaledUp > rgbMaxChannel ? rgbMaxChannel : scaledUp;
+}
+
+// Converts samples in the image from the given 'gAMA' value to the sRGB gamma curve.
+static void avifApplyGama(avifRGBImage * rgb, float gama)
+{
+    const float gamma = 1.f / gama;
+    assert(rgb->format == AVIF_RGB_FORMAT_RGBA || rgb->format == AVIF_RGB_FORMAT_RGB);
+    const uint32_t samplesPerPixel = rgb->format == AVIF_RGB_FORMAT_RGBA ? 4 : 3;
```

Comment:
`avifRGBFormatChannelCount()`  
`avifRGBImagePixelSize()` may be of some use too